src/Bridge/Doctrine/Orm/SubresourceDataProvider.php

```diff
@@ -195,4 +122,77 @@ public function getSubresource(string $resourceClass, array $identifiers, array
 
         return $context['collection'] ? $query->getResult() : $query->getOneOrNullResult();
     }
+
+    /**
+     * @throws RuntimeException
+     */
+    private function buildQuery(array $identifiers, array $context, QueryNameGenerator $queryNameGenerator, QueryBuilder $previousQueryBuilder, string $previousAlias, int $remainingIdentifiers, QueryBuilder $topQueryBuilder = null): QueryBuilder
+    {
+        if (!$remainingIdentifiers) {
+            return $previousQueryBuilder;
+        }
+
+        $topQueryBuilder = $topQueryBuilder ?? $previousQueryBuilder;
+
+        list($identifier, $identifierResourceClass) = $context['identifiers'][$remainingIdentifiers - 1];
+        $previousAssociationProperty = $context['identifiers'][$remainingIdentifiers][0] ?? $context['property'];
+
+        $manager = $this->managerRegistry->getManagerForClass($identifierResourceClass);
+
+        if (!$manager instanceof EntityManagerInterface) {
+            throw new RuntimeException("The manager for $identifierResourceClass must be an EntityManager.");
+        }
+
+        $classMetadata = $manager->getClassMetadata($identifierResourceClass);
+
+        if (!$classMetadata instanceof ClassMetadataInfo) {
+            throw new RuntimeException(
+                "The class metadata for $identifierResourceClass must be an instance of ClassMetadataInfo."
+            );
+        }
+
+        $qb = $manager->createQueryBuilder();
+        $alias = $queryNameGenerator->generateJoinAlias($identifier);
+        $relationType = $classMetadata->getAssociationMapping($previousAssociationProperty)['type'];
+        $normalizedIdentifiers = isset($identifiers[$identifier]) ? $this->normalizeIdentifiers(
+            $identifiers[$identifier],
+            $manager,
+            $identifierResourceClass
+        ) : [];
+
+        switch ($relationType) {
+            // MANY_TO_MANY relations need an explicit join so that the identifier part can be retrieved
+            case ClassMetadataInfo::MANY_TO_MANY:
+                $joinAlias = $queryNameGenerator->generateJoinAlias($previousAssociationProperty);
+
+                $qb->select($joinAlias)
+                    ->from($identifierResourceClass, $alias)
+                    ->innerJoin("$alias.$previousAssociationProperty", $joinAlias);
+
```

Comment:
for cosmetics, new line is useless here imo \o/

Reply:
Sure :)